### ISSUE TITLE
out-of-bouds read 

### ISSUE THREAD

**Issue by mlite** [cee-studio/orca#395]
I got this error from latest `bot-ping-pong.c`

```
Memory access error: reading from the outside of a memory block; abort!
  # Reading 1 bytes from 0xffe2215c will read undefined values.
  # 
  # The memory-block-to-be-read (start:0xffe2548c, size:1 bytes) is bound to '
                                                                              g�' at
  #    file:/workspace/orca/discord-websockets.cpp::941, 0
  # 
  #                               0xffe2548c           0xffe2548c
  #                               +--------------------------+
  #                         ......| memory-block-to-be-read  |
  #                               +--------------------------+
  #                         ^~~~~~~~~~
  # the read starts at 0xffe2215c that is 13104 bytes before the memory block start.
  # 
  # Stack trace (most recent call first) of the read.
  # [0]  file:/workspace/orca/json-actor.c::76, 3
  # [1]  file:/workspace/orca/json-actor.c::1291, 9
  # [2]  file:/workspace/orca/json-actor.c::1342, 12
  # [3]  file:/workspace/orca/json-actor.c::1470, 5
  # [4]  file:/workspace/orca/json-actor.c::1509, 23
  # [5]  file:/workspace/orca/discord-websockets.cpp::81, 13
  # [6]  file:/workspace/orca/json-actor.c::1209, 31
  # [7]  file:/workspace/orca/json-actor.c::1274, 19
  # [8]  file:/workspace/orca/json-actor.c::1350, 21
  # [9]  file:/workspace/orca/json-actor.c::1470, 5
  # [10]  file:/workspace/orca/json-actor.c::1509, 23
  # [11]  file:/workspace/orca/discord-websockets.cpp::430, 13
  # [12]  file:/workspace/orca/discord-websockets.cpp::458, 5
  # [13]  file:/workspace/orca/discord-websockets.cpp::722, 7
  # [14]  file:/workspace/orca/curl-websocket.c::587, 17
  # [15]  file:/workspace/orca/curl-websocket.c::795, 5
  # [16]  file:/workspace/orca/curl-websocket.c::814, 23
  # [17]  file:/workspace/packages/curl-7.72.0/lib/sendf.c::582, 15
  # [18]  file:/workspace/packages/curl-7.72.0/lib/sendf.c::666, 10
  # [19]  file:/workspace/packages/curl-7.72.0/lib/transfer.c::894, 26
  # [20]  file:/workspace/packages/curl-7.72.0/lib/transfer.c::1254, 14
  # [21]  file:/workspace/packages/curl-7.72.0/lib/multi.c::2150, 16
  # [22]  file:/workspace/packages/curl-7.72.0/lib/multi.c::2420, 14
  # [23]  file:/workspace/orca/discord-websockets.cpp::911, 13
  # [24]  file:/workspace/orca/discord-websockets.cpp::947, 5
  # [25]  file:/workspace/orca/discord-public.cpp::179, 3
  # [26]  file:/workspace/orca/bots/bot-ping-pong.cpp::57, 3
```

**Comment by mlite** [comment]
I'm looking at the problem now.

**Comment by mlite** [comment]
fixed in ff68509
